src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -31,34 +34,52 @@ private static void ProcessStartupHooks()
             }
 
             // Parse startup hooks variable
-            string[] startupHooks = startupHooksVariable.Split(Path.PathSeparator);
-            foreach (string startupHook in startupHooks)
+            string[] startupHookParts = startupHooksVariable.Split(Path.PathSeparator);
+            StartupHookNameOrPath[] startupHooks = new StartupHookNameOrPath[startupHookParts.Length];
+            for (int i = 0; i < startupHookParts.Length; i++)
             {
-                if (String.IsNullOrEmpty(startupHook))
+                string startupHookPart = startupHookParts[i];
+                if (string.IsNullOrEmpty(startupHookPart))
                 {
                     throw new ArgumentException(SR.Argument_InvalidStartupHookSyntax);
                 }
-                if (PathInternal.IsPartiallyQualified(startupHook))
+
+                if (Path.IsPathFullyQualified(startupHookPart))
+                {
+                    startupHooks[i].Path = startupHookPart;
+                }
+                else
                 {
-                    throw new ArgumentException(SR.Argument_AbsolutePathRequired);
```

Comment:
It might be nice for diagnostic purposes to continue throwing an ArgumentException for relative paths. Or, if we fall back to exceptions thrown by the AssemblyName constructor, we should update the error handling section of the spec.

Reply:
Now it's throwing ArgumentException for anything but simple names (and wraps exceptions from AssemblyName ctor).